File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/shape/UXShape.php
<?php
namespace php\gui\shape;

use php\gui\paint\UXColor;
use php\gui\paint\UXPaint;
use php\gui\UXNode;

/**
 * Class UXShape
 * @package php\gui\shape
 * @packages gui, javafx
 */
abstract class UXShape extends UXNode
{
    /**
     * @var bool
     */
    public $smooth = true;

    /**
     * @var UXPaint
     */
    public $fill;

    /**
     * @var UXPaint
     */
    public $stroke;

    /**
     * @var double
     */
    public $strokeWidth = 1;

    /**
     * @var string INSIDE, OUTSIDE, CENTERED
     */
    public $strokeType = 'OUTSIDE';
}

File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/shape/UXShape.java
package org.develnext.jphp.ext.javafx.classes.shape;

import javafx.scene.paint.Color;
import javafx.scene.paint.Paint;
import javafx.scene.shape.Shape;
import javafx.scene.shape.StrokeType;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import org.develnext.jphp.ext.javafx.classes.UXNode;
import php.runtime.annotation.Reflection.*;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Name(JavaFXExtension.NS + "shape\\UXShape")
public abstract class UXShape<T extends Shape> extends UXNode<Shape> {
    interface WrappedInterface {
        @Property boolean smooth();
        @Property double strokeWidth();
        @Property StrokeType strokeType();

        @Property Paint fill();
        @Property Paint stroke();
    }

    public UXShape(Environment env, T wrappedObject) {
        super(env, wrappedObject);
    }

    public UXShape(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Override
    public T getWrappedObject() {
        return (T) super.getWrappedObject();
    }
}
